service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java

```diff
@@ -73,4 +86,37 @@ private void checkForTrailingSlashes(ServiceContext serviceContext) {
     protected HttpProxyServiceHandler createHandler() {
         return new HttpProxyServiceHandler();
     }
+
+    /**
+     * Method returning true if overriding accept/connect options found
+     * @param serviceContext
+     * @return
+     */
+    private boolean noOverridingAcceptConnectOptions(ServiceContext serviceContext) {
+        AcceptOptionsContext acceptOptionsContext = serviceContext.getAcceptOptionsContext();
+        ConnectOptionsContext connectOptionsContext = serviceContext.getConnectOptionsContext();
+
+        boolean bindsEmpty = acceptOptionsContext.getBinds().isEmpty();
+        boolean hasAcceptTransport = hasTransport(acceptOptionsContext.asOptionsMap());
+        boolean hasConnectTransport = hasTransport(connectOptionsContext.asOptionsMap());
+
+        return bindsEmpty && !hasAcceptTransport && !hasConnectTransport;
+    }
+
+    /**
+     * Method returning whether option list has *.transport(s)
+     * @param map
+     * @return
+     */
+    private boolean hasTransport(Map<String, Object> map) {
```

Comment:
It seems like a hack to check ".transport" (but I don't have a good solution)
- if there is some option like "socks.transport" would break this check
- there may be tcp.bind option
- there may be layers of transport options

The best thing is get ResourceAddress for accept and connect and check. But I don't see an easy way to do that here. It would be nice to do the check here while gateway is starting.

May be this can be done during request processing (which is bit late but better late than never). Need to think more.